app/controllers/accounts_controller.rb

```diff
@@ -29,7 +29,7 @@ def show
         end
 
         if current_user.nil?
-          @pinned_statuses = cache_collection(@account.pinned_statuses.without_local_only, Status) if show_pinned_statuses?
+          @pinned_statuses = cache_collection(filtered_pinned_statuses.without_local_only, Status) if show_pinned_statuses?
```

Comment:
This scoped collection was already defined in the profile—it limits to just pinned statuses that have a visibility of public or unlisted, then the `without_local_only` scope further limits the collection.